gazebo_ros2_control/src/gazebo_system.cpp

```diff
@@ -181,52 +188,127 @@ control_toolbox::Pid GazeboSystem::extractPID(
-    min_integral_error = std::numeric_limits<double>::min();
-  }
+    if (joint_info.parameters.find(prefix + "min_integral_error") != joint_info.parameters.end()) {
+      min_integral_error = std::stod(joint_info.parameters.at(prefix + "min_integral_error"));
+    } else {
+      min_integral_error = std::numeric_limits<double>::min();
+    }
 
-  if (joint_info.parameters.find(prefix + "antiwindup") != joint_info.parameters.end()) {
-    if (joint_info.parameters.at(prefix + "antiwindup") == "true" ||
-      joint_info.parameters.at(prefix + "antiwindup") == "True")
-    {
-      antiwindup = true;
+    if (joint_info.parameters.find(prefix + "antiwindup") != joint_info.parameters.end()) {
+      if (joint_info.parameters.at(prefix + "antiwindup") == "true" ||
+        joint_info.parameters.at(prefix + "antiwindup") == "True")
+      {
+        antiwindup = true;
+      }
     }
+
+    RCLCPP_INFO_STREAM(
+      this->nh_->get_logger(),
+      "Setting kp = " << kp << "\t"
+                      << " ki = " << ki << "\t"
+                      << " kd = " << kd << "\t"
+                      << " max_integral_error = " << max_integral_error << "\t"
+                      << "antiwindup =" << std::boolalpha << antiwindup);
+
+    pid.initPid(kp, ki, kd, max_integral_error, min_integral_error, antiwindup);
   }
+  return are_pids_set;
+}
 
-  RCLCPP_INFO_STREAM(
-    this->nh_->get_logger(),
-    "Setting kp = " << kp << "\t"
-                    << " ki = " << ki << "\t"
-                    << " kd = " << kd << "\t"
-                    << " max_integral_error = " << max_integral_error);
 
-  return control_toolbox::Pid(kp, ki, kd, max_integral_error, min_integral_error, antiwindup);
+bool GazeboSystem::extractPIDFromParameters(
+  const std::string & control_mode, const std::string & joint_name,
+  control_toolbox::Pid & pid)
+{
+  const std::string parameter_prefix = "pid_gains." + control_mode + "." + joint_name;
+  auto get_pid_entry = [this, parameter_prefix](const std::string & entry, double & value) -> bool {
+      try {
+        // Check if the parameter is declared, if not, declare the default value NaN
+        if (!this->nh_->has_parameter(parameter_prefix + "." + entry)) {
+          this->nh_->declare_parameter<double>(
+            parameter_prefix + "." + entry,
+            std::numeric_limits<double>::quiet_NaN());
+        }
+        value = this->nh_->get_parameter(parameter_prefix + "." + entry).as_double();
+      } catch (rclcpp::exceptions::ParameterNotDeclaredException & ex) {
+        RCLCPP_ERROR(
+          this->nh_->get_logger(),
+          "Parameter '%s' not declared, with error %s", entry.c_str(), ex.what());
+        return false;
+      } catch (rclcpp::exceptions::InvalidParameterTypeException & ex) {
+        RCLCPP_ERROR(
+          this->nh_->get_logger(),
+          "Parameter '%s' has wrong type: %s", entry.c_str(), ex.what());
+        return false;
+      }
+      return std::isfinite(value);
```

Comment:
include `<cmath>`

Reply:
Sure. I will add it. I just wanted to leave it for initial review. Thank you for the review. I'll get it done soon